### PR TITLE
Functional test tx type 6 - Subtask #678 - Closes #935

### LINKED ISSUE #935: Functional tx test type 6 - Subtask #678
Parent #678

### PR DESCRIPTION
**Pull request by diego-G**
Subtask #678 
Closes #935

### CODE REVIEW

File: test/functional/http/post/6.dapps.inTransfer.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		});
+
+		describe('amount', function () {
+
+			it('using < 0 should fail', function () {
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, -1, node.gAccount.password);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('status').to.equal(400);
+					node.expect(res).to.have.nested.property('body.message').to.equal('Invalid transaction body - Failed to validate transaction schema: Value -1 is less than minimum 0');
```
Comment (Isabello) [inline test/functional/http/post/6.dapps.inTransfer.js]: Why are we allowing 0 lsk in transfers? The minimum should be 1, just like a type 0.

File: test/functional/http/post/6.dapps.inTransfer.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+						badTransactions.push(transaction);
+					});
+			});
+		});
+	});
+
+	describe('transactions processing', function () {
+
+		it('invented dapp id should fail', function () {
+			var inventedDappId  = '1';
```
Comment (Isabello) [inline test/functional/http/post/6.dapps.inTransfer.js]: maybe call this unknownDapp instead.

File: test/functional/shared.js [hunk 3]
```diff
@@ -104,6 +104,9 @@ function invalidAssets (account, option, badTransactions) {
 			case 'dapp':
 				transaction = node.lisk.dapp.createDapp(account.password, null, node.guestbookDapp);
 				break;
+			case 'inTransfer':
+				transaction = node.lisk.transfer.createInTransfer(node.guestbookDapp.id, Date.now(), node.gAccount.password);
+				break;
```
Comment (Isabello) [inline test/functional/shared.js]: the naming here is odd. The other cases go `node.lisk.NAME`

Here we are using the name of type 0 as the base name. Should it be called intransfer instead?

Comment (karmacoma) [reply test/functional/shared.js]: > The naming here is odd. The other cases go node.lisk.NAME.

I believe naming relates to how the `lisk-js` API is designed.

> Here we are using the name of type 0 as the base name. Should it be called intransfer instead?

This is a correct strategy, using parent/base type name, which all cases are.

File: test/functional/http/post/6.dapps.inTransfer.js (hunk 1 above)
Comment (karmacoma) [reply test/functional/http/post/6.dapps.inTransfer.js]: This the minimum defined by the parent transaction schema, where child types might only have a fee.

Then for an actual `transfer` it is checked here in verify: https://github.com/LiskHQ/lisk/blob/development/logic/transfer.js#L62

Schema checks are designed to be less concrete to allow for inheritance pattern.

**Pull request merged**